File: string/remove-trailing-zeros-from-a-string/remove-trailing-zeros-from-a-string.go
package removetrailingzerosfromastring

import "strings"

func removeTrailingZeros(num string) string {
	ei := len(num) - 1

	for {
		if num[ei] != '0' {
			break
		}
		ei--
	}

	return num[:ei+1]
}

func removeTrailingZerosEasy(num string) string {
	return strings.TrimRight(num, "0")
}

File: string/remove-trailing-zeros-from-a-string/remove-trailing-zeros-from-a-string.ts
function removeTrailingZeros(num: string): string {
    let i = num.length - 1;
    while (i) {
        if (num[i] !== "0") {
            break;
        }
        i--;
    }

    return num.slice(0, i + 1)
};

function removeTrailingZerosEasy(num: string): string {
    return num.replace(/0+$/, "")
}
